pkg/controller/generate-registry/generate.go

```diff
@@ -298,21 +298,38 @@ func checkChecksumCosign(pkgInfo *registry.PackageInfo, checksumAssetName string
 	if signatureAssetName == "" {
 		return nil
 	}
-	var certificateAssetName string
-	for _, suf := range []string{"-keyless.pem", ".pem"} {
-		if _, ok := assetNames[checksumAssetName+suf]; ok {
-			certificateAssetName = checksumAssetName + suf
+	var certificateAssetName, pubKeyAssetName string
+	for assetName := range assetNames {
+		if strings.HasSuffix(assetName, "cosign.pub") {
```

Comment:
```suggestion
		if assetName == "cosign.pub" {
```

Is `cosign.pub` suffix?

Reply:
It is sometimes used alone, but it's not unusual for it to be a suffix. There are a few in the registry already, and some more coming up from #32207 

```
$ grep -n "[^/]cosign.pub" registry.yaml 
9325:      #     - https://github.com/argoproj/argo-workflows/releases/download/{{.Version}}/argo-workflows-cosign.pub
51941:            - https://raw.githubusercontent.com/stackrox/kube-linter/refs/tags/{{.Version}}/kubelinter-cosign.pub
51964:            - https://raw.githubusercontent.com/stackrox/kube-linter/refs/tags/{{.Version}}/kubelinter-cosign.pub
51986:            - https://raw.githubusercontent.com/stackrox/kube-linter/refs/tags/{{.Version}}/kubelinter-cosign.pub
56802:              - https://github.com/twpayne/chezmoi/releases/download/{{.Version}}/chezmoi_cosign.pub
```

Reply:
I see. Thank you for your explanation.